annotations/annotators.py

```diff
@@ -100,15 +100,25 @@ def get_content(self, resource):
 
     def get_resource(self):
         """
-        Retrieve the resource represented by our :class:`.Text` instance.
+        Retrieve the resource represented by our :class:`.Text` instance, 
+        focusing on the tokenized content for word-based annotation.
         """
         if self.resource is not None:
             return self.resource
-        if not self.text.repository:
-            return
-        manager = self.text.repository.manager(self.context['user'])
-        self.resource = manager.content(id=int(self.text.repository_source_id))
-        return self.resource
+
+        if not self.text:
+            return None
+
+        # Directly retrieve the tokenized content from the Text model.
+        try:
+            # Retrieve the tokenized content of the text
+            resource = self.text.tokenizedContent
+            self.resource = resource
+            return self.resource
+        except Text.DoesNotExist:
```

Comment:
where is that being thrown? Isn't that thrown from a db transaction? Is there a db call made here?